nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -149,3 +221,64 @@ def get_statistic_points(self, model: TModel, graph: NNCFGraph) -> StatisticPoin
         :param graph: Model graph.
         :return: Statistic points, for which StatisticsCollector should collect statistics.
         """
+
+    def _get_activations(
+        self, dataset: Dataset, nodes_to_compress: List[NNCFNode], graph: NNCFGraph, model: TModel
+    ) -> Dict[str, List[np.ndarray]]:
+        """
+        Collects input activations for the given nodes on the dataset.
+
+        :param dataset: Dataset to collect values.
+        :param nodes_to_compress: List of nodes, whose inputs are collected.
+        :param model: Model for statistics collection.
+        :param graph: Model graph.
+        :return: statistics values itself per node name.
+        """
+        subset_size = 128
+
+        activations = {}
+        _collected_stat_inputs_map = {}
+        statistic_container = StatisticPointsContainer()
+        all_act_nodes = []
+        act_vs_shared_node_names_mapping = defaultdict(list)
+        from nncf.openvino.graph.metatypes.openvino_metatypes import OVMatMulMetatype
+
+        filtered_nodes = filter(lambda node: node.metatype == OVMatMulMetatype, nodes_to_compress)
+        for node in filtered_nodes:
+            act_node, output_port_id = self._get_activation_node_and_port(node, graph)
+            act_node_name = act_node.node_name
+            if act_node_name in all_act_nodes:
```

Comment:
To speed up algorithm, `all_act_nodes` should be a set

Reply:
corrected